### PR TITLE
kubeadm: use T.Run API in app/

### PR DESCRIPTION
**Pull request by bart0sh**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Used [T.Run API](https://golang.org/pkg/testing/#T.Run) for kubeadm tests in app/

This should improve testing output and make it more visible
which test is doing what.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/preflight/checks_test.go [hunk 1]
```diff
@@ -790,12 +793,14 @@ func TestNumCPUCheck(t *testing.T) {
...
 
 	for _, rt := range tests {
-		warnings, errors := NumCPUCheck{NumCPU: rt.numCPU}.Check()
-		if len(warnings) != rt.numWarnings {
-			t.Errorf("expected %d warning(s) but got %d: %q", rt.numWarnings, len(warnings), warnings)
-		}
-		if len(errors) != rt.numErrors {
-			t.Errorf("expected %d warning(s) but got %d: %q", rt.numErrors, len(errors), errors)
-		}
+		t.Run(strconv.Itoa(rt.numCPU), func(t *testing.T) {
```
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_test.go]: possibly add something else as part of the name here?
`fmt.Sprintf(number of CPUs: %d, rt.numCPU)`

or add proper `name` element as part of the `tests` struct.

**Review by neolit123 — COMMENTED** [review]
thanks @bart0sh 
added one minor comment.

/approve

**Comment by neolit123** [comment]
needs a hack/update-gofmt.sh
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/72483/pull-kubernetes-verify/117567/

File: cmd/kubeadm/app/constants/constants_test.go [hunk 2]
```diff
@@ -182,28 +186,30 @@ func TestEtcdSupportedVersion(t *testing.T) {
...
-		} else {
-			if rt.expectedError != nil {
-				t.Errorf("failed EtcdSupportedVersion:\n\texpected error: %v, but got no error", rt.expectedError)
-			} else if strings.Compare(actualVersion.String(), rt.expectedVersion.String()) != 0 {
-				t.Errorf(
-					"failed EtcdSupportedVersion:\n\texpected version: %s\n\t  actual version: %s",
-					rt.expectedVersion.String(),
-					actualVersion.String(),
-				)
+		t.Run(rt.kubernetesVersion, func(t *testing.T) {
```
Comment (chuckha) [inline cmd/kubeadm/app/constants/constants_test.go]: I might prefer giving these test cases names describing what they are testing, but i'm not terribly concerned about it

**Review by chuckha — APPROVED** [review]
This looks good to me, thank you! Only minor concern about naming semantics, but I don't really care so long as the test name is unique enough to figure out what failed.

File: cmd/kubeadm/app/constants/constants_test.go (hunk 2 above)
Comment (bart0sh) [reply cmd/kubeadm/app/constants/constants_test.go]: I think the test output looks quite understandable:

```
$ go test -run TestEtcdSupportedVersion
=== RUN   TestEtcdSupportedVersion
=== RUN   TestEtcdSupportedVersion/1.99.0
=== RUN   TestEtcdSupportedVersion/1.10.2
=== RUN   TestEtcdSupportedVersion/1.11.0
=== RUN   TestEtcdSupportedVersion/1.12.1
=== RUN   TestEtcdSupportedVersion/1.13.1
=== RUN   TestEtcdSupportedVersion/1.14.0
--- PASS: TestEtcdSupportedVersion (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.99.0 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.10.2 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.11.0 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.12.1 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.13.1 (0.00s)
    --- PASS: TestEtcdSupportedVersion/1.14.0 (0.00s)
```

Adding something like "testing etc supported version 1.10.2" wouldn't add much content here in my opinion.

Comment (chuckha) [reply cmd/kubeadm/app/constants/constants_test.go]: Yep, the output looks good. I like to think the name is more about what is this test case actually testing, or the assertion being made. The name "1.14.0" doesn't mean much to me, but I could certainly trace it back and figure it out which is more than enough imo.

Comment (bart0sh) [reply cmd/kubeadm/app/constants/constants_test.go]: great! lgtm then :) ?

**Comment by chuckha** [comment]
/lgtm

thanks for the clean up!

**Pull request merged**